helix-term/src/commands.rs

```diff
@@ -4499,22 +4499,19 @@ pub mod insert {
     use helix_core::auto_pairs;
 
     pub fn insert_char(cx: &mut Context, c: char) {
-        let (view, doc) = current!(cx.editor);
-
-        let hooks: &[Hook] = match cx.editor.config.auto_pairs {
-            true => &[auto_pairs::hook, insert],
-            false => &[insert],
-        };
-
+        let (view, doc) = current_ref!(cx.editor);
         let text = doc.text();
         let selection = doc.selection(view.id);
+        let auto_pairs = cx.editor.get_document_auto_pairs(doc.id());
 
-        // run through insert hooks, stopping on the first one that returns Some(t)
-        for hook in hooks {
-            if let Some(transaction) = hook(text, selection, c) {
-                doc.apply(&transaction, view.id);
-                break;
-            }
```

Comment:
Any reason for the changes here? This was intended to be a point where we could inject additional hooks in the future.

Reply:
I also just realized it's not possible to opt out of auto pairs anymore: before setting `auto_pairs = false` would disable this entirely, but now language specific settings are always applied even if the global config is none.